code/modules/reagents/reagent_containers/borghydro.dm

```diff
@@ -56,6 +56,13 @@
 			if(RG.total_volume < RG.maximum_volume) 	//Don't recharge reagents and drain power if the storage is full.
 				R.cell.use(charge_cost) 					//Take power from borg...
 				RG.add_reagent(reagent_ids[mode], 5)		//And fill hypo with reagent.
+			else                                            //if active mode is full loop through the list and fill the first one that is not full
+				for(var/i in 1 to reagent_list.len)            
```

Comment:
Best to use a foreach loop here. In DM it has this syntax:
```js
for(var/i in list)
  var/type/path/R = i
  i.do_stuff()
```
You can also include the path in `for(var/type/path/R in list)`. But this will filter the list on the type you give in the select. This will be slower than the way I showed before but will ensure that all entries in the loop are of that type. If you are sure they are of one type then you can best use the first way presented. Else you can use the filter option.

Reply:
I've overlooked the fact that the borg hypo has multiple `datum/reagents`. With a s. Due to this the `id` of the reagent in question is unknown. Getting it from the reagents holder will work untill it is empty.
The for loop here is a valid way to solve this issue in combination with line 64 where you get the id from the list.